server/src/apub/shared_inbox.rs

```diff
@@ -7,14 +7,10 @@ use crate::{
   apub::{
```

Comment:
This is by far our worst file in the codebase. At some point we could use your expertise on re-arranging / de-duplicating a lot of work in here.